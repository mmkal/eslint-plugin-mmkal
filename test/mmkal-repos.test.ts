import * as fs from 'fs'
import * as path from 'path'
import {expect, test} from 'vitest'

const reposToTest = [
  'https://github.com/mmkal/expect-type', //
  'https://github.com/mmkal/trpc-cli',
]

for (const repo of reposToTest) {
  test(
    `lint ${repo}`,
    async () => {
      const {execa} = await import('execa')
      await execa('pnpm', ['build'])
      const tmp = path.join(`/tmp/eslint-plugin-mmkal-testing`, Date.now().toString())
      fs.mkdirSync(tmp, {recursive: true})
      await execa('git', ['clone', repo], {cwd: tmp})
      const [cloneName, ...otherFiles] = fs.readdirSync(tmp)
      const clone = path.join(tmp, cloneName)
      expect(otherFiles).toEqual([])
      await execa('pnpm', ['install'], {cwd: clone})
      await execa(path.join(process.cwd(), 'node_modules', '.bin', 'link'), [process.cwd()], {cwd: clone})
      const {all: lint} = await execa('pnpm', ['eslint', '.', '--fix', '--max-warnings', '0'], {cwd: clone})

      expect(lint!.toLowerCase()).not.toContain('error')
    },
    {timeout: 30_000},
  )
}
