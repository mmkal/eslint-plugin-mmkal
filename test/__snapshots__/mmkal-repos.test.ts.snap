// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lint https://github.com/mmkal/expect-type 1`] = `
"
<dir>/expect-type/src/branding.ts
  <line>:<col>  error  The \`Function\` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type  @typescript-eslint/no-unsafe-function-type

<dir>/expect-type/src/index.ts
  <line>:<col>  error  The \`Function\` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type  @typescript-eslint/no-unsafe-function-type

<dir>/expect-type/src/utils.ts
  <line>:<col>  error  'secret' is assigned a value but only used as a type. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

<dir>/expect-type/test/deprecations.test.ts
  <line>:<col>  warning  Use default import for module \`node:path\`  unicorn/import-style

<dir>/expect-type/test/ts-output.ts
  <line>:<col>  warning  Use default import for module \`node:path\`  unicorn/import-style

<dir>/expect-type/test/types.test.ts
  <line>:<col>  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions
  <line>:<col>  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions

<dir>/expect-type/test/usage.test.ts
  <line>:<col>  error  'A' is defined but only used as a type. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  <line>:<col>  error  'B' is defined but only used as a type. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  <line>:<col>  error  'C' is defined but only used as a type. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars

✖ 10 problems (8 errors, 2 warnings)
"
`;

exports[`lint https://github.com/mmkal/trpc-cli 1`] = `
"
<dir>/trpc-cli/src/index.ts
  <line>:<col>  error  Use \`Array#toSorted()\` instead of \`Array#sort()\`  unicorn/no-array-sort
  <line>:<col>   error  Use \`Array#toSorted()\` instead of \`Array#sort()\`  unicorn/no-array-sort

<dir>/trpc-cli/src/json-schema.ts
  <line>:<col>  error  Use \`Array#toSorted()\` instead of \`Array#sort()\`  unicorn/no-array-sort
  <line>:<col>   error  Use \`Array#toSorted()\` instead of \`Array#sort()\`  unicorn/no-array-sort

<dir>/trpc-cli/src/standard-schema/errors.ts
  <line>:<col>  error  Use \`Array#toSorted()\` instead of \`Array#sort()\`  unicorn/no-array-sort

✖ 5 problems (5 errors, 0 warnings)
"
`;
